phasmophobia_evidence_v3/widget.js

```diff
@@ -677,7 +724,19 @@ const resetGhost = (newName, state) => {
 };
 
 const resetName = (newName, state) => {
-  state.ghostName = newName;
+  fullName = "";
+  if (newName) {
+    if (config.nameStrings.autoCapitalize) {
+      nameSplit = newName.split(" ");
+      for (let i = 0; i < nameSplit.length; i++) {
+        nameSplit[i] = nameSplit[i][0].toUpperCase() + nameSplit[i].substr(1).toLowerCase();
+      }
+      fullName = nameSplit.join(" ");
+    } else {
+      fullName = newName;
+    }
+  }
+  state.ghostName = fullName;
```

Comment:
This feels repeated a lot between the three methods. Specifically the capitalization piece.

Every time we separate the word by whitespace into an array, then iterate through, uppercase, then rejoin.

I think we should move that out into a method that does the following:

```
// Takes a string and returns a string with each character of each word capitalized
const camelCase = (sentence) = > {
  return sentence.replace(/(^\w{1})|(\s+\w{1})/g, letter => letter.toUpperCase());
}
```

This uses a regular expression that should capture the first letter of every word and capitalize it.

Reply:
Changes were implemented in Update 26fc4a7